internal/base_component.go

```diff
@@ -62,7 +62,7 @@ func (base *BaseComponent) SetName(name string) {
 	base.compName = name
 }
 
-func (base *BaseComponent) Configure() error {
+func (base *BaseComponent) Configure(validateCredentials bool) error {
```

Comment:
validateCredential does not make sense for all other components, so better to call this something which says "its running in background/daemon mode"

Reply:
done